maven/bnd-run-maven-plugin/src/main/java/aQute/bnd/maven/run/plugin/RunMojo.java

```diff
@@ -161,15 +175,20 @@ private Operation getOperation() {
 							if (watchedPaths.contains(path) && semaphore.tryAcquire()) {
 								scheduledExecutor.schedule(() -> {
 									try {
-										logger.info("Detected change to {}. Reloading.", path);
+										logger.info("Detected change to {}. {}.", path, resolve ? "Resolving" : "Reloading");
+
+										if (resolve) {
+											pool.submit(() -> resolve(run, pl, watchedPaths, watcher));
```

Comment:
A potential issue here is multiple resolve operations concurrently since the watcher can wake up and see a change and start a new resolve. Perhaps you should serialize resolve operations so only one per run at a time?

Reply:
I think I've got a good solution to this now.